src/Base/Identifier.php

```diff
@@ -19,23 +19,42 @@
  */
 class Identifier extends ValueObject
 {
-    public function __construct(UuidInterface $identifier)
-    {
-        parent::__construct($identifier);
-    }
+    /**
+     * @var UuidInterface
+     */
+    private $value;
 
-    public function equals(ValueObject $identifier): bool
+    /**
+     * Identifier constructor.
+     * @param UuidInterface $identifier
+     */
+    public function __construct(UuidInterface $identifier)
     {
-        return (string) $this == (string) $identifier;
+        $this->value = $identifier;
     }
 
+    /**
+     * @return Identifier
+     */
     public static function generate(): self
     {
         return new static(Uuid::uuid1());
     }
 
+    /**
+     * @param string $identifier
+     * @return Identifier
+     */
     public static function fromString(string $identifier): self
     {
         return new static(Uuid::fromString($identifier));
     }
+
+    /**
+     * @return string
+     */
+    public function __toString(): string
```

Comment:
To avoid duplication in value objects of one property, you could create something like SinglePropertyValueObject

```php
class SinglePropertyValueObject extends ValueObject
{
    /**
     * @var mixed
     */
    private $value;

    public function __construct($value)
    {
        $this->value = $value;
    }

    public function __toString(): string
    {
        return $this->value;
    }
}
```